myriad-scheduler/src/main/java/com/ebay/myriad/api/ClustersResource.java

```diff
@@ -96,6 +101,48 @@ public Response flexUp(FlexUpClusterRequest request) {
 
     @Timed
     @PUT
+    @Path("/flexupservice")
+    @Produces(MediaType.TEXT_PLAIN)
+    @Consumes(MediaType.APPLICATION_JSON)
+    public Response flexUpservice(FlexUpServiceRequest request) {
+      Preconditions.checkNotNull(request,
+              "request object cannot be null or empty");
+
+      LOGGER.info("Received Flexup a Service Request");
+
+      Integer instances = request.getInstances();
+      String profile = request.getServiceName();
+
+      LOGGER.info("Instances: {}", instances);
+      LOGGER.info("Profile: {}", profile);
+      
+      // Validate profile request
+      Response.ResponseBuilder response = Response.status(Response.Status.ACCEPTED);
+      
+      if (cfg.getServiceConfiguration(profile) == null) {
+        response.status(Response.Status.BAD_REQUEST)
+                .entity("Service does not exist: " + profile);
+        LOGGER.error("Provided service does not exist " + profile);
+        return response.build();
+      }
+      
+      if (!this.isValidInstanceSize(instances)) {
+        response.status(Response.Status.BAD_REQUEST)
+                .entity("Invalid instance size: " + instances);
+        LOGGER.error("Invalid instance size request " + instances);
+        return response.build();
+      }
+
+      try {
+        this.myriadOperations.flexUpAService(instances, profile);
+      } catch (MyriadBadConfigurationException e) {
```

Comment:
> then we flexed up 2 more they will start since validation only checks for instances > 0
Shouldn't the validation prevent that from happening? For e.g. when NMs are being flexed down, we LOG a warn message if number of NMs requested to flex down is greater than the number of NMs previously flexed up (for a specific profile).
https://github.com/mesos/myriad/pull/129/files#diff-f2ece694fc6140ea94557a9bfaffe8d4R111

> It is easier to make checks in a single place like MyriadOperations and not in two places: ClustersResource and Main
Agreed.